### PR TITLE
Support for attribute selectors and nthChild queries other than interger

### PR DESCRIPTION
**Pull request by Starzu**

### CODE REVIEW

File: core/src/test/scala/scalacss/RandomData.scala [hunk 1]
```diff
@@ -34,16 +34,12 @@ object RandomData {
...
-    val needInt = NonEmptyList[Int => Pseudo](
-      NthChild, NthLastChild, NthLastOfType, NthOfType)
-
     val needStr = NonEmptyList[String => Pseudo](
       Custom(_, PseudoClass), Custom(_, PseudoElement), Lang, Not(_))
 
     lazy val self: Gen[Pseudo] =
       Gen.frequency[Pseudo](
         objects.size -> Gen.oneofL(objects),
-        needInt.size -> Gen.oneofL(needInt).flatMap(Gen.positiveint.map),
```
Comment (japgolly) [inline core/src/test/scala/scalacss/RandomData.scala]: What you've done here is completely remove `NthChild, NthLastChild, NthLastOfType, NthOfType` from generatable RandomData.

Could you please add new generators for `NthChild, NthLastChild, NthLastOfType, NthOfType, Attr, AttrExists, Attr...`

**Comment by japgolly** [comment]
@Starzu This is mostly looking pretty good so thanks again. As always there's the slightly tedious task of updating the documentation too. Could you please do so? If you haven't seen it yet, it's online at http://japgolly.github.io/scalacss/book/ and the [Developing](http://japgolly.github.io/scalacss/book/dev.html) page has instructions for how to work with it locally.

File: core/src/test/scala/scalacss/RandomData.scala (hunk 1 above)
Comment (japgolly) [inline core/src/test/scala/scalacss/RandomData.scala]: Btw, I've just updated `RandomData.scala` so make sure you do a merge or rebase before commencing work. Cheers.

**Comment by Starzu** [comment]
@japgolly Take a look again, please.

**Comment by japgolly** [comment]
Nice work, thanks. Merged.

**Comment by japgolly** [comment]
FYI I made some finishing touches in f580e2a58c105614199fbf9d2e4a81155a5bc7df

**Pull request merged**